### ISSUE TITLE
False positive with jsx-pascal-case

### ISSUE THREAD

**Issue by Hypnosphi** [yannickcr/eslint-plugin-react#2683]
```jsx
<H1>Hello!</H1>

// error  Imported JSX component H1 must be in PascalCase  react/jsx-pascal-case
```

Introduced in 7.20.1, probably caused by #2636

Workaround: use `allowAllCaps` or `ignore` option

**Comment by ljharb** [comment]
cc @yacinehmito 

I'll have a fix ready shortly.

**Comment by ljharb** [comment]
Thanks for the repro; I'll have the fix released later today.

**Comment by ljharb** [comment]
v7.20.2 is released.
